certora/specs/ERC20Wrapper.spec

```diff
@@ -0,0 +1,190 @@
+import "helpers.spec"
+import "ERC20.spec"
+
+methods {
+    underlying()                       returns(address) envfree
+    underlyingTotalSupply()            returns(uint256) envfree
+    underlyingBalanceOf(address)       returns(uint256) envfree
+    underlyingAllowanceToThis(address) returns(uint256) envfree
+
+    depositFor(address, uint256)       returns(bool)
+    withdrawTo(address, uint256)       returns(bool)
+    recover(address)                   returns(uint256)
+}
+
+use invariant totalSupplyIsSumOfBalances
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Helper: consequence of `totalSupplyIsSumOfBalances` applied to underlying                                           │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+function sumOfUnderlyingBalancesLowerThanUnderlyingSupply(address a, address b) {
```

Comment:
Can we call this helper `requireSumOf...`?

It's very important to make it explicit that it's adding an assumption.

Reply:
we can, but its also important to notice that this assumption is a direct consequence of the `totalSupplyIsSumOfBalances` invariant on `underlying`. So its not an additional assumption

Reply:
I made it return a bool, so that you can do

`require sumOfUnderlyingBalancesLowerThanUnderlyingSupply(a, b);`